src/AzureAuth/PublicClientAuth.cs

```diff
@@ -12,27 +13,50 @@ namespace Microsoft.Authentication.AzureAuth
     /// <summary>
     /// A class for handling AAD Token acuisition, results logging, and telemetry collection.
     /// </summary>
-    public class AuthOrchestrator
+    public class PublicClientAuth : IPublicClientAuth
     {
         private readonly ILogger logger;
         private readonly IEnv env;
         private readonly ITelemetryService telemetryService;
-        private readonly ITokenFetcher tokenFetcher;
+        private readonly IMsalWrapper tokenFetcher;
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="AuthOrchestrator"/> class.
+        /// Initializes a new instance of the <see cref="PublicClientAuth"/> class.
         /// </summary>
         /// <param name="logger">An <see cref="ILogger"/>.</param>
         /// <param name="env">An <see cref="IEnv"/>.</param>
         /// <param name="telemetryService">An <see cref="ITelemetryService"/>.</param>
-        /// <param name="tokenFetcher">An <see cref="ITokenFetcher"/>.</param>
+        /// <param name="tokenFetcher">An <see cref="IMsalWrapper"/>.</param>
         /// <exception cref="ArgumentNullException">All parameters must not be null.</exception>
-        public AuthOrchestrator(ILogger logger, IEnv env, ITelemetryService telemetryService, ITokenFetcher tokenFetcher)
+        public PublicClientAuth(ILogger logger, IEnv env, ITelemetryService telemetryService, IMsalWrapper tokenFetcher)
         {
             this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
             this.env = env ?? throw new ArgumentNullException(nameof(env));
             this.telemetryService = telemetryService ?? throw new ArgumentNullException(nameof(telemetryService));
             this.tokenFetcher = tokenFetcher ?? throw new ArgumentNullException(nameof(tokenFetcher));
         }
+
+        /// <inheritdoc/>
+        public TokenResult Token(Guid client, Guid tenant, IEnumerable<string> scopes, AuthMode[] authModes, string domain, string prompt, TimeSpan timeout)
+        {
+            var result = this.tokenFetcher.AccessToken(
+                this.logger,
+                client,
+                tenant,
+                scopes,
+                authModes.Combine().PreventInteractionIfNeeded(this.env),
+                domain,
+                PromptHint.Prefixed(prompt),
+                timeout);
+
+            result.Attempts.SendTelemetry(this.telemetryService);
+
+            if (result.Success != null)
+            {
+                return result.Success.TokenResult;
+            }

```

Comment:
[nit]: Can we reverse this? Kind of like returning `result.Success.TokenResult` in the end. So

`if (result.Success == null)
            {
                return null;
            }

            return result.Success.TokenResult;`

Reply:
Yep, we can do that!